### ISSUE TITLE
Shellcheck does not complain about find -exec + (invalid on AIX)

### ISSUE THREAD

**Issue by bmarwell** [koalaman/shellcheck#2737]
#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/any/interpreter
find $HOME -name "*.temp" -exec rm -f '{}' +
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

`'-exec ... +' terminaltion is specific for GNU utilities and not available on e.g. AIX`

**Comment by tstenner** [comment]
On the one hand this would be nice. On the other hand, the OS utilities aren't part of the shell and AIX is missing lots of basic functionality (`-v` for `rm` and `cp`, `-r` for `ln`, basically everything for `sed`, ...) that's used widely.

**Comment by bmarwell** [comment]
Exactly. I wouldn't know which shell script linter to use for this, if not shellcheck. 🙈🤷🏻‍♂️

**Comment by koalaman** [comment]
`-exec ... +` is actually a POSIX standard, not a GNU specific feature. I definitely feel your pain trying to write AIX compatible scripts and I would love to be able to check against e.g. AT&T UNIX System 3 compatibility, but unfortunately this is way too granular, so POSIX compatibility is the lowest baseline available at this time.

**Comment by bmarwell** [comment]
Thanks, found it:

> `-exec  utility_name  [argument ...] ;`
> `-exec  utility_name  [argument ...]   {} +`
>    The end of the primary expression shall be punctuated by a semicolon or by a plus sign.

source: https://pubs.opengroup.org/onlinepubs/007904975/utilities/find.html

I will open a case with IBM then.
